components/com_users/tmpl/profile/default_custom.php

```
@@ -56,17 +56,15 @@
                             <?php echo $field->title; ?>
                         </dt>
                         <dd>
-                            <?php if (array_key_exists($field->fieldname, $customFields)) : ?>
-                                <?php echo strlen($customFields[$field->fieldname]->value) ? $customFields[$field->fieldname]->value : Text::_('COM_USERS_PROFILE_VALUE_NOT_FOUND'); ?>
-                            <?php elseif (HTMLHelper::isRegistered('users.' . $field->id)) : ?>
-                                <?php echo HTMLHelper::_('users.' . $field->id, $field->value); ?>
-                            <?php elseif (HTMLHelper::isRegistered('users.' . $field->fieldname)) : ?>
-                                <?php echo HTMLHelper::_('users.' . $field->fieldname, $field->value); ?>
-                            <?php elseif (HTMLHelper::isRegistered('users.' . $field->type)) : ?>
-                                <?php echo HTMLHelper::_('users.' . $field->type, $field->value); ?>
-                            <?php else : ?>
-                                <?php echo HTMLHelper::_('users.value', $field->value); ?>
-                            <?php endif; ?>
+                            <?php
+                            foreach ([$field->id, $field->fieldname, $field->type, 'value'] as $key) {
+                                try {
+                                    echo HTMLHelper::_('users.' . $key, $field->value);
+                                    break;
+                                } catch (\InvalidArgumentException $e) {
```

The `HTMLHelper::_` throws InvalidArgumentException at a few places, so here you catch all of them, not only the one you want to catch. So you will hide other errors, too, with that code.

---

Would it be better if I revert this part and manually register the function with HTMLHelper before it is called?